source/Pacioli/Pacioli.WebApi/Controllers/UserController.cs

```diff
@@ -18,39 +20,77 @@ namespace Pacioli.WebApi.Controllers
     public class UserController : ControllerBase
     {
         private readonly UserManager<User> _userManager;
+        private readonly RoleManager<IdentityRole> _roleManager;
         private readonly IConfiguration _configuration;
+        private readonly AccessTokenGenerator _accessTokenGenerator;
 
-        public UserController(UserManager<User> userManager, IConfiguration configuration)
+        public UserController(UserManager<User> userManager, IConfiguration configuration, RoleManager<IdentityRole> roleManager)
         {
             _configuration = configuration;
+            _roleManager = roleManager;
             _userManager = userManager;
+            //TODO : Move to DI.
+            _accessTokenGenerator = new AccessTokenGenerator();
+        }
+
+        //TODO : Registration should require a special one-time permission token generated by staff.
+        [HttpPost("register")]
+        public async Task<IActionResult> RegisterAsync(RegisterModel model)
+        {
+            if (ModelState.IsValid is false) 
+                return BadRequest();
+
+            bool userExists = await _userManager.FindByEmailAsync(model.Email) is not null ||
+                              await _userManager.FindByNameAsync(model.Username) is not null;
+            if (userExists)
+                return Conflict();
+
+            var user = new User
+            {
+                UserName = model.Username,
+                Email = model.Email,
+                SecurityStamp = Guid.NewGuid().ToString()
+            };
+
+            var userCreationResult = await _userManager.CreateAsync(user, model.Password);
+            if (userCreationResult.Succeeded is false)
+                return StatusCode(StatusCodes.Status500InternalServerError);
```

Comment:
I think this will return false if the password does not qualify password conditions ie password is only 4 characters. this shouldn't return a 500 status code, right? Shouldn't it return errors? What do you think?

Reply:
I agree, perhaps this should return `BadRequest();`